### DISCUSSION TITLE
how to convert png from indexed to rgb

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#5475] [Help] [answered]
How do I do this? I tried "magick mogrify -colorspace rgb *.png" but that didn't work.

**Comment by fmw42** [comment, marked as answer]
In convert or magick, you just add PNG24: to the output file as PNG24:output.suffix. In mogrify try

 `magick mogrify -type truecolor *.png`

**Reply by ghost** [reply]
That didn't work. Now they just all have a black background and are still indexed.

**Reply by fmw42** [reply]
Post an example input png.   That command worked for me.  Do your PNGs have transparency? If so, then you likely need to use -type truecoloralpha

What is your exact IM version?

**Reply by ghost** [reply]
![0](https://user-images.githubusercontent.com/69117768/186065208-9dff9aca-cc3a-4564-b312-887c052c5615.png)
version 7.1.0-6

**Reply by fmw42** [reply]
It has transparency. So use -type truecoloralpha in place of -type truecolor

**Reply by ghost** [reply]
okay thanks

**Reply by fmw42** [reply]
Your image is already 32-bit color. It is 8-bits per channel with 4 channels RGBA

```
 Format: PNG (Portable Network Graphics)
  Mime type: image/png
  Class: DirectClass
  Geometry: 170x167+0+0
  Units: Undefined
  Colorspace: sRGB
  Type: TrueColorAlpha
  Base type: Undefined
  Endianness: Undefined
  Depth: 8-bit
  Channel depth:
    red: 8-bit
    green: 8-bit
    blue: 8-bit
    alpha: 8-bit

```

**Reply by fmw42** [reply]
Looks like the PNG color-type on your image is still indexed even though IM identify says truecolor rather than palette. So try

`magick mogrify -define png:color-type=2 *.png`

That seems to change it for me

**Reply by ghost** [reply]
I don't know why but that destroys the images for me. All the smoothness is gone.

**Reply by fmw42** [reply]
My mistake. It should have been

`magick mogrify -define png:color-type=6 *.png`
